_studio/shared/umc/codec/h265_dec/src/umc_h265_slice_decoding.cpp

```diff
@@ -317,6 +386,122 @@ int H265Slice::getNumRpsCurrTempList() const
+    {
+        if (getRPS()->getUsed(i))
+        {
+            RefPicPOCSetLtCurr[NumPicLtCurr] = getRPS()->getPOC(i);
+            NumPicLtCurr++;
+        }
+    }
+    m_SliceHeader.m_numPicLtCurr = NumPicLtCurr;
+
+    H265PicParamSet const* pps = GetPicParam();
+    VM_ASSERT(pps);
+
+    int32_t numPicTotalCurr = NumPicStCurr0 + NumPicStCurr1 + NumPicLtCurr;
+    if (pps->pps_curr_pic_ref_enabled_flag)
+        numPicTotalCurr++;
+
+    int32_t rpsPOCCurrList1[MAX_NUM_REF_PICS + 1];
+
+    int32_t cIdx = 0;
+    int32_t rIdx = 0;
+    for (i = 0; i < NumPicStCurr0; cIdx++, i++)
+    {
+        m_SliceHeader.m_RpsPOCCurrList0[cIdx] = RefPicPOCSetStCurr0[i];
+    }
+    for (i = 0; i < NumPicStCurr1; cIdx++, i++)
+    {
+        m_SliceHeader.m_RpsPOCCurrList0[cIdx] = RefPicPOCSetStCurr1[i];
+    }
+    for (i = 0; i < NumPicLtCurr; cIdx++, i++)
+    {
+        m_SliceHeader.m_RpsPOCCurrList0[cIdx] = RefPicPOCSetLtCurr[i];
+    }
+    if (pps->pps_curr_pic_ref_enabled_flag)
+        m_SliceHeader.m_RpsPOCCurrList0[cIdx++] = m_SliceHeader.m_poc;
+
+    if (m_SliceHeader.slice_type == B_SLICE)
+    {
+        cIdx = 0;
+        for (i = 0; i < NumPicStCurr1; cIdx++, i++)
+        {
+            rpsPOCCurrList1[cIdx] = RefPicPOCSetStCurr1[i];
+        }
+        for (i = 0; i < NumPicStCurr0; cIdx++, i++)
+        {
+            rpsPOCCurrList1[cIdx] = RefPicPOCSetStCurr0[i];
+        }
+        for (i = 0; i < NumPicLtCurr; cIdx++, i++)
+        {
+            rpsPOCCurrList1[cIdx] = RefPicPOCSetLtCurr[i];
+        }
+        if (pps->pps_curr_pic_ref_enabled_flag)
+            rpsPOCCurrList1[cIdx++] = m_SliceHeader.m_poc;
+    }
+
+    RefPicListModification &refPicListModification = m_SliceHeader.m_RefPicListModification;
+
+    for (rIdx = 0; rIdx <= m_SliceHeader.m_numRefIdx[0] - 1; rIdx++)
+    {
+        cIdx = refPicListModification.ref_pic_list_modification_flag_l0 ? refPicListModification.list_entry_l0[rIdx] : rIdx % numPicTotalCurr;
+        m_SliceHeader.m_RefPOCList[0][rIdx] = m_SliceHeader.m_RpsPOCCurrList0[cIdx];
+    }
+
+    int32_t const numRpsCurrTempList0 = (std::max)(m_SliceHeader.m_numRefIdx[0], numPicTotalCurr);
+    if (pps->pps_curr_pic_ref_enabled_flag &&
+        !refPicListModification.ref_pic_list_modification_flag_l0 &&
+        numRpsCurrTempList0 > m_SliceHeader.m_numRefIdx[0])
+    {
+        m_SliceHeader.m_RefPOCList[0][m_SliceHeader.m_numRefIdx[0] - 1] = m_SliceHeader.m_poc;
+    }
+
+    if (m_SliceHeader.slice_type == P_SLICE)
+    {
+        m_SliceHeader.m_numRefIdx[1] = 0;
+    }
+    else
+    {
+        for (rIdx = 0; rIdx <= m_SliceHeader.m_numRefIdx[1] - 1; rIdx++)
+        {
+            cIdx = refPicListModification.ref_pic_list_modification_flag_l1 ? refPicListModification.list_entry_l1[rIdx] : rIdx % numPicTotalCurr;
+            m_SliceHeader.m_RefPOCList[1][rIdx] = rpsPOCCurrList1[cIdx];
```

Comment:
I start to fix myself as static analyzer:). But it seems that you can use `rpsPOCCurrList1` uninitialized here. You actually fill out this array above if and only if `m_SliceHeader.slice_type == B_SLICE`, but you will enter this block of code both for I_SLICE and B_SLICE. For I_SLICE it will be unitialized. Should you update condition and avoid this code execution for the I_SLICE? or maybe you should add the whole function code protection to avoid execution for I_SLICE?

Reply:
if (sliceHdr->slice_type != I_SLICE)
{
            pSlice->setRefPOCListSliceHeader();
}
Because the code above, I_Slice will not enter setRefPOCListSliceHeader function.

Reply:
Ok, I see. However, I don't think that current solution is the best one. I think you need to do one of 2 things:
1. Either move `if (sliceHdr->slice_type != I_SLICE)` check inside the function in a form like:
```
void H265Slice::setRefPOCListSliceHeader()
{
    if (m_SliceHeader.slice_type == I_SLICE)
         return;
    ...
}
```

2. or add assert to check that setRefPOCListSliceHeader really works with the correct expected slice type.

I strongly recommend to follow 1st solution...

Reply:
IMHO, do not think it is best to move if (sliceHdr->slice_type != I_SLICE) inside function, because in its parent function decodeSlice, many operations have dependency w/ slice type, all of them check slice type in decodeSlice, it is more readable. I can add one more check inside function, but has code redundancy concern.

Reply:
And I hove code maintenance concern. Someone will need to add yet another call to this function in some other place and will forget to add a check for I_SLICE to prevent the function call. If you want to minimize impact - add assert. It is specifically designed for that purpose: notify that something impossible had happenned:
```
void H265Slice::setRefPOCListSliceHeader()
{
    assert(m_SliceHeader.slice_type != I_SLICE);
    ...
}

```

Reply:
@guowillyy : I see your change in the beginning of the function. It's ok with me. Thank you for the fix.